documentation/internal/Assignment-Process.md

```diff
@@ -0,0 +1,57 @@
+# Assignment Process
+
+The logic of assigning a review spans between both web app and server, front end is only concerned with queries and mutations to review_assignment and review_question_assignment tables, whereas back end does most of the grunt work (through actions and triggers) and generates records in those tables based on user permissions and states of application and it's review.
+
+## Schema
+
+Seems like a good starting (anchor) point for describing this functionality, let's have look:
+
+![Review Assignment](images/Review-Assignment.png)
+
+First of all, some bulletpoint (and rules) to help clarify the diagram
+
+- One review assignment should exists per reviewer/application/stage/level
+- Review assignment can have the following statuses
+  - `Available` -> Can be assigned by assigner
+  - `Self-assigned by another` -> Was available for self assignment (but self assigned by someone else)
+  - `Assigned` -> At least on review_question_assignment was linked to review_assignment
+  - `Available for self-assignment` -> Can be self assigned
+- Review assignments can be restricted by section (template_section_restriction, which is an array of section codes)
```

Comment:
Maybe clarify here that "restrictions" means the sections that you *can* do, but if there's not restrictions specified, they can do *all* sections.

Reply:
5d80e7de004763c37418086f20a9e44f0ee99f87